File: IsItASquare/solve/cpp/IsItASquareSolver.cpp
#include <fstream>
#include <topcoder>
#include "IsItASquare.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
    try {
        std::ifstream ifs(argv[1]);
        vector<int> x; tc::read(ifs, x); tc::next(ifs);
        vector<int> y; tc::read(ifs, y);
        ifs.close();

        std::ofstream ofs(argv[2]);
        IsItASquare solver;
        tc::show(ofs, solver.isSquare(x, y));
        ofs.close();
    } catch (std::exception &e) {
        std::cerr << e.what() << std::endl;
    }
    return 0;
}

File: IsItASquare/solve/java/IsItASquareSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class IsItASquareSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<Integer> xBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] x = new int[xBoxed.size()];
        for (int _i = 0; _i < xBoxed.size(); ++_i)
            x[_i] = xBoxed.get(_i);
        reader.next();
        
        List<Integer> yBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] y = new int[yBoxed.size()];
        for (int _i = 0; _i < yBoxed.size(); ++_i)
            y[_i] = yBoxed.get(_i);
        reader.close();

        IsItASquare solver = new IsItASquare();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.isSquare(x, y));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }    
}
